scenes/RgbPalette.shader

```diff
@@ -0,0 +1,22 @@
+shader_type canvas_item;
+
+// the replacement colors
+uniform vec4 red : hint_color;
+uniform vec4 green : hint_color;
+uniform vec4 blue : hint_color;
+uniform vec4 black : hint_color;
+
+void fragment() {
```

Comment:
Add a class-level description of what this shader is all about. It's crazy.

Reply:
Added verbose description. I agree, it is quite crazy. Maybe we should publish this on the godot forums.